### PR TITLE
Add "fly-speed" flag.

### PR DESCRIPTION
**Pull request by eth-p**
Similar to walk-speed, but for players who are flying.

Just a small change that is convenient for creative servers with large spawn areas.

### CODE REVIEW

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	    
+	protected FlySpeedFlagHandler(Plugin plugin, Session session)
+	{
+		super(plugin, session);
+	}
+	
+	@Override
+    public void initialize(Player player, Location current, ApplicableRegionSet set)
+	{
+		Double speed = WorldGuardUtils.queryValue(player, current.getWorld(), set.getRegions(), Flags.WALK_SPEED);
```
Comment (aromaa) [inline WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java:47]: FLY_SPEED

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		this.handleValue(player, speed);
+	}
+	
+	@Override
+	public boolean onCrossBoundary(Player player, Location from, Location to, ApplicableRegionSet toSet, Set<ProtectedRegion> entered, Set<ProtectedRegion> exited, MoveType moveType)
+	{
+		Double speed = WorldGuardUtils.queryValue(player, to.getWorld(), toSet.getRegions(), Flags.WALK_SPEED);
```
Comment (aromaa) [inline WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java:54]: FLY_SPEED

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/wrappers/WorldGuardCommunicator.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package net.goldtreeservers.worldguardextraflags.wg.wrappers;
 
+import net.goldtreeservers.worldguardextraflags.wg.handlers.*;
```
Comment (aromaa) [inline WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/wrappers/WorldGuardCommunicator.java:3]: I don't really like to use the wildcard

**Review by aromaa — CHANGES_REQUESTED** [review]
Theres also few formatting errors. Would be nice to combine walk speed and fly speed to abstract class as it has the same logic.

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java (hunk 1 above)
Comment (eth-p) [reply WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java:47]: Thanks... Copy paste blindness, sorry.

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/wrappers/WorldGuardCommunicator.java (hunk 3 above)
Comment (eth-p) [reply WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/wrappers/WorldGuardCommunicator.java:3]: My bad. My IDE automatically collapses multiple imports into wildcards, and I didn't have it turned off when opening the WGEF project. I'll fix that in the next commit.

**Comment by eth-p** [comment]
> Theres also few formatting errors.

Will fix that, sorry. I generally try to mimic the projecy style, but I miss space/tab differences from time to time.

> Would be nice to combine walk speed and fly speed to abstract class as it has the same logic.

I was originally considering that, but I wasn't sure of the best place to put the abstract class. Would the same package be fine, or do you have a package/sourceset for abstract and interface code?

**Comment by eth-p** [comment]
I couldn't find the formatting errors you mentioned, but I did manage to fix the issues and refactor the handler logic into an abstract class.

I even added a bonus "spectate-speed" flag that takes precedence over the "fly-speed" flag if the player is in spectator mode.

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/flags/Flags.java [hunk 4]
```diff
@@ -32,6 +32,8 @@
 	public final static SetFlag<String> CONSOLE_COMMAND_ON_EXIT = WorldGuardUtils.getCommunicator().getCustomSetFlag("console-command-on-exit", new CommandStringFlag(null));
 	
 	public final static DoubleFlag WALK_SPEED = new DoubleFlag("walk-speed");
+	public final static DoubleFlag FLY_SPEED = new DoubleFlag("fly-speed");
+	public final static DoubleFlag SPECTATE_SPEED = new DoubleFlag("spectate-speed");
```
Comment (aromaa) [inline WG/src/main/java/net/goldtreeservers/worldguardextraflags/flags/Flags.java:36]: I think this is overkill, adds bloat and is really case specific

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import net.goldtreeservers.worldguardextraflags.wg.wrappers.HandlerWrapper;
+
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+
+public abstract class AbstractSpeedFlagHandler extends HandlerWrapper
+{
+    
+    private Float originalSpeed;
```
Comment (aromaa) [inline WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java:22]: Formatting

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	
+	protected abstract void setSpeed(Player player, float speed);
+
+	protected DoubleFlag getFlag(Player player, Location from, Location to, ApplicableRegionSet toSet, Set<ProtectedRegion> entered, Set<ProtectedRegion> exited, MoveType moveType)
+	{
+		return this.flag;	
+	}
+	
+	@Override
+    public void initialize(Player player, Location current, ApplicableRegionSet set)
```
Comment (aromaa) [inline WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java:42]: Formatting

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+	{
+		Double speed = WorldGuardUtils.queryValue(player, to.getWorld(), toSet.getRegions(), this.getFlag(player, from, to, toSet, entered, exited, moveType));
+		this.handleValue(player, speed);
+		return true;
+	}
+	
+	
+	private void handleValue(Player player, Double speed)
+	{
+        if (speed != null)
```
Comment (aromaa) [inline WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java:59]: Formatting

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+public class FlySpeedFlagHandler extends AbstractSpeedFlagHandler
+{
+	public static final Factory FACTORY(Plugin plugin)
+	{
+		return new Factory(plugin);
+	}
+	
+    public static class Factory extends HandlerWrapper.Factory<FlySpeedFlagHandler>
+    {
+        public Factory(Plugin plugin)
```
Comment (aromaa) [inline WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java:30]: Formatting

**Review by aromaa — CHANGES_REQUESTED** [review]
Marked some formatting errors now also

File: WG/src/main/java/net/goldtreeservers/worldguardextraflags/flags/Flags.java (hunk 4 above)
Comment (eth-p) [reply WG/src/main/java/net/goldtreeservers/worldguardextraflags/flags/Flags.java:36]: Alright. Will be dropped in my next push.

**Comment by eth-p** [comment]
Thanks for pointing out the formatting issues for me. After looking at the other files in the project, it appears that a few of them are either spaces, tabs, or a mix of the two. That would probably explain why my IDE failed to select the correct indentation method 😛

Basing the indentation it off your last commit, I replaced all the spaces from the changed files with tabs.
I also removed the spectator-speed flag as requested.

**Review by aromaa — APPROVED** [review]
Thanks. Looks fine now

**Pull request merged**